core/src/test/scala/org/apache/spark/sql/delta/MergeIntoSuiteBase.scala

```diff
@@ -933,8 +933,7 @@ abstract class MergeIntoSuiteBase
 
       append(Seq((100, 100), (3, 5)).toDF("key2", "value"))
       // cache is in effect, as the above change is not reflected
-      checkAnswer(spark.table(s"delta.`$tempPath`"),
-        Row(2, 2) :: Row(1, 4) :: Row(100, 100) :: Row(3, 5) :: Nil)
+      checkAnswer(spark.table(s"delta.`$tempPath`"), Row(2, 2) :: Row(1, 4) :: Nil)
```

Comment:
what does this change have to do with error subclasses?

Reply:
this change was already checked into master ... was it failing?

Reply:
It's a fix that was in 3.3.1. Just changed this for the tests to pass (again will revert before merging). See https://github.com/delta-io/delta/commit/529313d61a0d998b66b6fa0cad7d35e971edfb94